### PR TITLE
Feature/IO_669_Login_Screen

### PR DESCRIPTION
**Pull request by elizaveta-seleznyova-idf**

### CODE REVIEW

**Review by alexei-kruk-idf — APPROVED** [review]
Look at what comments I left for Egor and correct yourself

File: packages/presentation/lib/app/app_bloc.dart [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
   factory AppBloc() => _AppBloc();
 
   void handleRemoveRouteSettings(RouteSettings value);
+
+  void onItemTapped(int index);
 }
 
 class _AppBloc extends BlocImpl implements AppBloc {
   final _appData = AppData.init();
+  int selectedIndex = 0;
```
Comment (yuliya-klescheva-idf) [inline packages/presentation/lib/app/app_bloc.dart:20]: private

File: packages/presentation/lib/screen/login/login_screen.dart [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+import 'package:presentation/navigation/base_arguments.dart';
+import 'package:presentation/navigation/base_page.dart';
+import 'package:presentation/screen/login/login_bloc.dart';
+import 'package:presentation/screen/login/login_data.dart';
+import 'package:presentation/utils/image_path.dart';
+
+class LoginScreenArguments extends BaseArguments {
+  LoginScreenArguments({
+    Function(dynamic value)? result,
+  }) : super(result: result);
```
Comment (yuliya-klescheva-idf) [inline packages/presentation/lib/screen/login/login_screen.dart]: short super constructor

File: packages/presentation/lib/screen/profile/profile_screen.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import 'package:presentation/base/bloc_screen.dart';
+import 'package:presentation/navigation/base_arguments.dart';
+import 'package:presentation/navigation/base_page.dart';
+import 'package:presentation/screen/profile/profile_bloc.dart';
+
+class ProfileScreenArguments extends BaseArguments {
+  ProfileScreenArguments({
+    Function(dynamic value)? result,
+  }) : super(result: result);
```
Comment (yuliya-klescheva-idf) [inline packages/presentation/lib/screen/profile/profile_screen.dart]: short super

File: packages/presentation/lib/app/app_bloc.dart (hunk 1 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/app/app_bloc.dart:20]: why? In private class?

File: packages/presentation/lib/screen/login/login_screen.dart (hunk 2 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/login/login_screen.dart]: corrected

File: packages/presentation/lib/screen/profile/profile_screen.dart (hunk 3 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/profile/profile_screen.dart]: corrected

**Review by yuliya-klescheva-idf — APPROVED** [review]

File: android/app/build.gradle [hunk 4]
```diff
@@ -22,11 +22,14 @@ if (flutterVersionName == null) {
...
 apply plugin: 'com.android.application'
+// START: FlutterFire Configuration
+apply plugin: 'com.google.gms.google-services'
+// END: FlutterFire Configuration
 apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion flutter.compileSdkVersion
+    compileSdkVersion 33
```
Comment (sergei-mikhailovskii-idf) [inline android/app/build.gradle]:
```suggestion
    compileSdkVersion flutter.compileSdkVersion
    
```

File: android/app/build.gradle [hunk 5]
```diff
@@ -47,7 +50,7 @@ android {
         applicationId "com.example.movies_and_stars"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion flutter.minSdkVersion
+        minSdkVersion 21
```
Comment (sergei-mikhailovskii-idf) [inline android/app/build.gradle]:
```suggestion
        minSdkVersion flutter.minSdkVersion
```

File: android/app/build.gradle [hunk 6]
```diff
@@ -68,4 +71,5 @@ flutter {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.facebook.android:facebook-login:latest.release"
```
Comment (sergei-mikhailovskii-idf) [inline android/app/build.gradle]: specify version instead of `latest.release`

File: android/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">movies_and_stars</string>
+    <string name="facebook_app_id">3295496017387708</string>
+    <string name="fb_login_protocol_scheme">fb3295496017387708</string>
+    <string name="facebook_client_token">f5be1d4cac59b988743eec42b097a303</string>
```
Comment (sergei-mikhailovskii-idf) [inline android/app/src/main/res/values/strings.xml]: hide keys as Katya did it

File: android/build.gradle [hunk 8]
```diff
@@ -7,6 +7,9 @@ buildscript {
 
     dependencies {
         classpath 'com.android.tools.build:gradle:7.1.2'
+        // START: FlutterFire Configuration
+        classpath 'com.google.gms:google-services:4.3.10'
+        // END: FlutterFire Configuration
```
Comment (sergei-mikhailovskii-idf) [inline android/build.gradle]: remove comments

File: ios/Runner.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -3,16 +3,18 @@
 	archiveVersion = 1;
```
Comment (sergei-mikhailovskii-idf) [inline ios/Runner.xcodeproj/project.pbxproj:3]: ignore file which is higher plz

File: lib/firebase_options.dart [hunk 10]
```diff
@@ -0,0 +1,70 @@
+// File generated by FlutterFire CLI.
+// ignore_for_file: lines_longer_than_80_chars, avoid_classes_with_only_static_members
+import 'package:firebase_core/firebase_core.dart' show FirebaseOptions;
```
Comment (sergei-mikhailovskii-idf) [inline lib/firebase_options.dart]: seems that we don't need this file

File: packages/data/lib/repository/auth_repository.dart [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    final usersCollection = firebaseFirestore.collection(_firebaseCollection);
+    final userCheck = await usersCollection
+        .where('login', isEqualTo: user.login)
+        .where('password', isEqualTo: user.password)
+        .get();
+    return userCheck.docs.isNotEmpty;
+  }
+
+  @override
+  Future<UserEmailPass?> loginWithFaceBook() async {
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/repository/auth_repository.dart]: why optional?

File: packages/data/lib/repository/auth_repository.dart [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+    firebaseAuth.signInWithCredential(facebookAuthCredential);
+    return UserEmailPass(
+      user['email'] ?? '',
+      user['id'] ?? '',
+    );
+  }
+
+  @override
+  Future<UserEmailPass?> loginWithGoogle() async {
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/repository/auth_repository.dart]: -//-

File: packages/presentation/lib/enum /bottom_navigation_page_type.dart [hunk 13]
```diff
@@ -0,0 +1,6 @@
+enum BottomNavigationPageType{
+  home(),
+  ticket(),
+  notifications(),
+  profile();
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/enum /bottom_navigation_page_type.dart]: why did u add rounded brackets

File: packages/presentation/lib/screen/home/home_bloc.dart [hunk 14]
```diff
@@ -5,7 +5,7 @@ import 'package:domain/enum/movie_type.dart';
 import 'package:domain/model/movie_response.dart';
 import 'package:domain/use_case/get_movies_use_case.dart';
 import 'package:presentation/base/bloc.dart';
-import 'package:presentation/screen/home/enum/tab_state.dart';
+import 'package:presentation/enum%20/tab_state.dart';
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screen/home/home_bloc.dart]: seems that enum folder has space in the end. remove it

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: android/app/build.gradle (hunk 4 above)
Comment (elizaveta-seleznyova-idf) [reply android/app/build.gradle]: corrected

File: android/build.gradle (hunk 8 above)
Comment (elizaveta-seleznyova-idf) [reply android/build.gradle]: removed

File: android/app/build.gradle (hunk 6 above)
Comment (elizaveta-seleznyova-idf) [reply android/app/build.gradle]: add version

File: android/app/build.gradle (hunk 5 above)
Comment (elizaveta-seleznyova-idf) [reply android/app/build.gradle]: Screenshot 2022-10-12 at 19 18 12

File: lib/firebase_options.dart (hunk 10 above)
Comment (elizaveta-seleznyova-idf) [reply lib/firebase_options.dart]: why? we use it in presentation

File: packages/presentation/lib/enum /bottom_navigation_page_type.dart (hunk 13 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/enum /bottom_navigation_page_type.dart]: mystery

File: packages/presentation/lib/screen/home/home_bloc.dart (hunk 14 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/screen/home/home_bloc.dart]: removed

File: android/app/build.gradle (hunk 5 above)
Comment (sergei-mikhailovskii-idf) [reply android/app/build.gradle]: search where flutter.minSdkVersion is set and change it here

File: android/app/src/main/res/values/strings.xml (hunk 7 above)
Comment (sergei-mikhailovskii-idf) [reply android/app/src/main/res/values/strings.xml]: still actual

File: ios/Runner.xcodeproj/project.pbxproj (hunk 9 above)
Comment (sergei-mikhailovskii-idf) [reply ios/Runner.xcodeproj/project.pbxproj:3]: still actual

File: lib/firebase_options.dart (hunk 10 above)
Comment (sergei-mikhailovskii-idf) [reply lib/firebase_options.dart]: we can initialize Firebase without options. try to remove it from Firebase init and if everything works delete the file

File: packages/data/lib/repository/auth_repository.dart (hunk 11 above)
Comment (sergei-mikhailovskii-idf) [reply packages/data/lib/repository/auth_repository.dart]: still actual

File: ios/Runner.xcodeproj/project.pbxproj (hunk 9 above)
Comment (elizaveta-seleznyova-idf) [reply ios/Runner.xcodeproj/project.pbxproj:3]: deleted podfile.lock

File: packages/data/lib/repository/auth_repository.dart (hunk 12 above)
Comment (elizaveta-seleznyova-idf) [reply packages/data/lib/repository/auth_repository.dart]: corrected

File: packages/data/lib/repository/auth_repository.dart (hunk 11 above)
Comment (elizaveta-seleznyova-idf) [reply packages/data/lib/repository/auth_repository.dart]: corrected

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**